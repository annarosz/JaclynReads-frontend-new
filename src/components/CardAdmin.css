.cardAdmin {
  background-color: white;
  margin-bottom: 20px;
  display: flex;
  padding: 50px;
}

.cardAdminLeft {
  width: 150px;
  /* background-color: pink; */
  display: flex;
  align-items: center;
  flex-direction: column;
}

.cardAdminLeftTop {
  height: 200px;
  /* background-color: pink; */
}

.cardAdminLeftBottom {
  /* background-color: yellow; */
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.cardAdminRight {
  padding-left: 50px;
  max-width: 700px;
}

div.cardAdminLeftTop > img {
  max-width: 100%;
  max-height: 100%;
}

/* Media queries –––––––––––––––––––––––––––––––––––– */

@media only screen and (max-width: 550px) {

  .cardAdmin {
    flex-direction: column;
    padding: 20px;
  }

  .cardAdminRight {
    padding-left: 0px;
  }

}
