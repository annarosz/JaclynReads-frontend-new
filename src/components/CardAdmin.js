import React from "react";
import './CardAdmin.css';

function CardAdmin(props) {

  return (
    <div className="cardAdmin">

      <div className="cardAdminLeft">
        <div className="cardAdminLeftTop">
          <img src={props.url} alt={props.title}  ></img>
          </div>
        <div className="cardAdminLeftBottom">
          <button onClick={() => props.handleEditClick(props.reviewObject)} className="edit"> Edit</button>
          <button onClick={() => props.handleDeleteClick(props.title)} className="edit"> Delete</button>
        </div>
      </div>

      <div className="cardAdminRight">
        <p><strong>Title:</strong> {props.title}</p>
        <p><strong>Author:</strong> {props.author}</p>
        <p><strong>Review:</strong> {props.review}</p>
        <p><strong>Publisher:</strong> {props.publisher}</p>
        <p><strong>Year Published:</strong> {props.yearPublished}</p>
        <p><strong>Genre:</strong> {props.genre}</p>
        <p><strong>ISBN:</strong> {props.isbn}</p>
        <p><strong>URL to buy:</strong> {props.url}</p>
        <p><strong>SEO Keywords:</strong> {props.keywords}</p>
      </div>

    </div>
  );
}

export default CardAdmin;