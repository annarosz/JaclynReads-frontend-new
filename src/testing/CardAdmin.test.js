import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";
import CardAdmin from "../components/CardAdmin";

describe("<CardAdmin />", () => {
  const reviewList = [
    {
      title: "Three Little Pigs",
      author: "Big Bad Wolf",
      _id: "5d19cd74a0c3e2c7f68ff800",
      url: "https://bookmarks-rag.s3.amazonaws.com/bookmarks-content.jpeg"
    },
    {
      title: "Go Blues",
      author: "Cheer Squad",
      _id: "5d19cd74a0c3e2c7f68ff801",
      url: "https://bookmarks-rag.s3.amazonaws.com/bookmarks-content.jpeg"
    },
    {
      title: "Tim Tams are Awesome",
      author: "Robert De Matteo",
      _id: "5d19cd74a0c3e2c7f68ff802",
      url: "https://bookmarks-rag.s3.amazonaws.com/bookmarks-content.jpeg"
    }
  ];

  it("displays loading when no review in list", () => {
    const review = [];
    const wrapper = shallow(<CardAdmin data={review} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders CardAdmin component ", () => {
    const wrapper = shallow(<CardAdmin />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders .cardLeft ", () => {
    const wrapper = shallow(<CardAdmin />);
    expect(wrapper.find(".cardLeft")).toMatchSnapshot();
  });

  it("displays title in title field and author in author filed in CardAdmin.js", () => {
    const wrapper = render(
      <CardAdmin title={reviewList[0].title} author={reviewList[0].author} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("contains the review title", () => {
    const wrapper = mount(<CardAdmin title={reviewList[0].title} />);
    const text = wrapper.find(".cardAdminRight").text();
    expect(text.includes("Three")).toEqual(true);
  });

  it("calls the edit method on click", () => {
    const spy = sinon.spy();
    const wrapper = mount(
      <CardAdmin
        handleEditClick={() => {
          spy();
        }}
      />
    );

    wrapper.find(".edit").simulate("click");
    expect(spy.calledOnce).toBe(true);
  });

  it("calls the delete method on click", () => {
    const spy = sinon.spy();
    const wrapper = mount(
      <CardAdmin
        handleDeleteClick={() => {
          spy();
        }}
      />
    );

    wrapper.find(".delete").simulate("click");
    expect(spy.calledOnce).toBe(true);
  });
});
