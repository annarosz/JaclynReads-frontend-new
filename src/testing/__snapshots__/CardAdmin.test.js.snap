// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardAdmin /> displays loading when no review in list 1`] = `
<div
  className="cardAdmin"
>
  <div
    className="cardAdminLeft"
  >
    <div
      className="cardAdminLeftTop"
    >
      <img />
    </div>
    <div
      className="cardAdminLeftBottom"
    >
      <button
        className="edit"
        onClick={[Function]}
      >
         Edit
      </button>
      <button
        className="delete"
        onClick={[Function]}
      >
         Delete
      </button>
    </div>
  </div>
  <div
    className="cardAdminRight"
  >
    <p>
      Title: 
    </p>
    <p>
      Author: 
    </p>
    <p>
      Review: 
    </p>
    <p>
      Publisher: 
    </p>
    <p>
      Year Published: 
    </p>
    <p>
      Genre: 
    </p>
    <p>
      ISBN: 
    </p>
    <p>
      URL to buy: 
    </p>
    <p>
      SEO Keywords: 
    </p>
  </div>
</div>
`;

exports[`<CardAdmin /> displays title in title field and author in author filed in CardAdmin.js 1`] = `
<div
  class="cardAdmin"
>
  <div
    class="cardAdminLeft"
  >
    <div
      class="cardAdminLeftTop"
    >
      <img
        alt="Three Little Pigs"
      />
    </div>
    <div
      class="cardAdminLeftBottom"
    >
      <button
        class="edit"
      >
         Edit
      </button>
      <button
        class="delete"
      >
         Delete
      </button>
    </div>
  </div>
  <div
    class="cardAdminRight"
  >
    <p>
      Title: Three Little Pigs
    </p>
    <p>
      Author: Big Bad Wolf
    </p>
    <p>
      Review: 
    </p>
    <p>
      Publisher: 
    </p>
    <p>
      Year Published: 
    </p>
    <p>
      Genre: 
    </p>
    <p>
      ISBN: 
    </p>
    <p>
      URL to buy: 
    </p>
    <p>
      SEO Keywords: 
    </p>
  </div>
</div>
`;

exports[`<CardAdmin /> renders .cardLeft  1`] = `null`;

exports[`<CardAdmin /> renders CardAdmin component  1`] = `
<div
  className="cardAdmin"
>
  <div
    className="cardAdminLeft"
  >
    <div
      className="cardAdminLeftTop"
    >
      <img />
    </div>
    <div
      className="cardAdminLeftBottom"
    >
      <button
        className="edit"
        onClick={[Function]}
      >
         Edit
      </button>
      <button
        className="delete"
        onClick={[Function]}
      >
         Delete
      </button>
    </div>
  </div>
  <div
    className="cardAdminRight"
  >
    <p>
      Title: 
    </p>
    <p>
      Author: 
    </p>
    <p>
      Review: 
    </p>
    <p>
      Publisher: 
    </p>
    <p>
      Year Published: 
    </p>
    <p>
      Genre: 
    </p>
    <p>
      ISBN: 
    </p>
    <p>
      URL to buy: 
    </p>
    <p>
      SEO Keywords: 
    </p>
  </div>
</div>
`;
